src/sync/sync_session.cpp

```diff
@@ -514,6 +579,43 @@ bool SyncSession::wait_for_upload_completion_blocking()
     return false;
 }
 
+uint64_t SyncSession::register_progress_notifier(std::function<SyncProgressNotifierCallback> notifier,
+                                                 NotifierType direction,
+                                                 bool is_streaming)
+{
+    std::function<void()> invocation;
+    uint64_t token_value = 0;
+    {
+        std::lock_guard<std::mutex> lock(m_progress_notifier_mutex);
+        bool skip_registration = false;
```

Comment:
This should be moved down to immediately prior to the call to `create_notifier_invocation`.